diff --git a/src/wl/sys/wl_linux.c b/src/wl/sys/wl_linux.c
index 0d05100..4be2189 100644
--- a/src/wl/sys/wl_linux.c
+++ b/src/wl/sys/wl_linux.c
@@ -1643,7 +1643,11 @@ wl_ioctl(struct net_device *dev, struct ifreq *ifr, int cmd)
 		goto done2;
 	}
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 9, 0)
 	if (segment_eq(get_fs(), KERNEL_DS))
+#else
+	if (uaccess_kernel())
+#endif
 		buf = ioc.buf;
 
 	else if (ioc.buf) {